static/compiler.js

```diff
@@ -385,6 +400,7 @@ define(function (require) {
         status.toggleClass('error', failed);
         status.toggleClass('warning', warns);
         status.parent().attr('title', allText);
+        this.optButton.toggleClass("hidden", !result.hasOptOutput);
```

Comment:
I think I'd probably prefer it to be disabled rather than hidden. Having the UI move around when you pick a new compiler is a little odd, plus the `title` text for the button can give a clue as to how to enable it. Aids discoverability, else non-clang users won't know it's there. ("Show optimization output (clang-like compilers only)" or similar)